homeassistant/components/sensor/recorder.py

```diff
@@ -678,7 +678,9 @@ def validate_statistics(
 
     for state in sensor_states:
         entity_id = state.entity_id
-        state_class = state.attributes.get(ATTR_STATE_CLASS)
+        state_class: SensorStateClass | None = None
+        with contextlib.suppress(ValueError):
+            state_class = SensorStateClass(state.attributes.get(ATTR_STATE_CLASS))  # type: ignore[arg-type]
```

Comment:
#87082 has been approved
```suggestion
        state_class = try_parse_enum(SensorStateClass, state.attributes.get(ATTR_STATE_CLASS))
```